ala-name-matching-builder/src/main/java/au/org/ala/names/index/TaxonConceptInstance.java

```diff
@@ -1070,8 +1080,11 @@ private TaxonConceptInstance getResolvedAccepted(TaxonConceptInstance original,
     }
 
     private TaxonConceptInstance getResolved(TaxonConceptInstance original, int steps) {
-        if (steps <= 0)
+        if (steps <= 0) {
+            if (this.isForbidden())
+                return null;
             throw new ResolutionException("Detected possible loop resolving " + original);
```

Comment:
Does throw get executed? Already returned?

Reply:
Yes. Forbidden is a state that only a few TCIs have.